src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java

```diff
@@ -61,6 +63,34 @@ public void should_read_blob_and_save_metadata_in_database_when_zip_contains_met
         testBlobFileProcessed();
     }
 
+    @Test
+    public void should_continue_processing_blob_when_envelope_stuck_in_created_status() throws Exception {
+        uploadToBlobStorage(SAMPLE_ZIP_FILE_NAME, zipDir("zipcontents/ok"));
+
+        testBlobFileProcessed();
+
+        // pretend it is stuck
+        envelopeRepository.saveAll(
+            envelopeRepository
+                .findAll()
+                .stream()
+                .peek(envelope -> {
+                    envelope.setStatus(CREATED);
+                    envelope.setZipDeleted(false);
+                })
```

Comment:
please use `.map`

Reply:
java doesn't live in immutable world, but yep, whatever ;)